core/src/main/java/com/linecorp/armeria/common/util/AbstractListenable.java

```diff
@@ -33,17 +36,23 @@
     @SuppressWarnings("rawtypes")
     private static final Consumer[] EMPTY_LISTENERS = new Consumer[0];
 
+    @GuardedBy("reentrantLock")
     private final Set<Consumer<? super T>> updateListeners =
             new ObjectLinkedOpenCustomHashSet<>(IdentityHashStrategy.of());
 
+    private final ReentrantLock reentrantLock = new ReentrantLock();
+
     /**
      * Notify the new value changes to the listeners added via {@link #addListener(Consumer)}.
      */
     protected final void notifyListeners(T latestValue) {
         final Consumer<? super T>[] updateListeners;
-        synchronized (this.updateListeners) {
+        lock();
```

Comment:
Could you inline this method so that we don't see the errorprone warning, please?